Makefile

```diff
@@ -1,31 +1,36 @@
 .DEFAULT_GOAL := all
 sources = pydantic tests docs/plugins
 
+.PHONY: .pdm
+.pdm:  ## Check that PDM is isntallced
+	@pdm -V || echo 'Please install PDM: https://pdm.fming.dev/latest/\#installation'
+
+.PHONY: .pre-commit
+.pre-commit:  ## Check that PDM is isntallced
+	@pre-commit -V || echo 'Please install pre-commit: https://pre-commit.com/'
+
 .PHONY: install
-install:
-	python -m pip install -U pip
-	pip install -r requirements/all.txt
-	pip install -r requirements/pyproject-all.txt
-	pip install -e .
+install: .pdm .pre-commit
+	pdm install --group :all --dev
+	pre-commit install --install-hooks
 
 .PHONY: refresh-lockfiles
-refresh-lockfiles:       ## Sync lockfiles with requirements files.
-	bash requirements/refresh.sh
-
+refresh-lockfiles: .pdm       ## Sync lockfiles with requirements files.
+	pdm lock --refresh --group :all --dev
 
 .PHONY: rebuild-lockfiles
-rebuild-lockfiles:       ## Rebuild lockfiles from scratch, updating all dependencies
-	bash requirements/rebuild.sh
+rebuild-lockfiles: .pdm       ## Rebuild lockfiles from scratch, updating all dependencies
+	pdm lock --group :all --dev
 
 .PHONY: format
 format:
-	black $(sources)
-	ruff --fix $(sources)
+	pdm run black $(sources)
```

Comment:
does this mean you have to do `pdm run <command>` any time you want to run something in your dev environment? I guess maybe you are supposed to activate the environment with pdm or whatever and then it's not necessary?

Does the `--no-isolation` flag used below cause it to not create an environment and just use whatever is active?

Reply:
You can still activate the environment and then just run the thing. But `pdm run` does that for you.

Reply:
> Does the --no-isolation flag used below cause it to not create an environment and just use whatever is active?

I believe so. I guess we'll see when that cron job CI file runs.